geode-assembly/build.gradle

```diff
@@ -29,10 +27,17 @@ extraArchive {
   tests = false
 }
 
-// Gradle doesn't automatically remove the jar artifact even though we disabled it
-// this causes publishing to fail.  So we nuke all the disabled artifacts from all configurations.
+logger.info("Gradle doesn't automatically remove the jar artifact even though we disabled it")
+logger.info("this causes publishing to fail.  So we nuke all the disabled artifacts from all configurations.")
 configurations.all {
-  artifacts.removeAll artifacts.findAll { it instanceof ArchivePublishArtifact && !it.archiveTask.enabled }
+  def configName = it.name
+  artifacts.each {
+    logger.info("Pruning configName = " + configName + " name=" + it.name + " extension=" + it.extension + " classifier=" + it.classifier)
+  }
+  artifacts.removeAll{ it.name == jar.baseName && it.extension == 'jar' && it.classifier == jar.classifier }
```

Comment:
Is one of these conditions sufficient?

Reply:
No, or else you could remove a mis-named artifact, or the `tar` or `zip` artifacts, or accidentally add a `test` or `integrationTest` artifact